pkg/ioutils/fswriters.go

```diff
@@ -109,6 +117,12 @@ func (w *atomicFileWriter) Close() (retErr error) {
 			return err
 		}
 	}
+
+	info, statErr := w.f.Stat()
+	if statErr == nil {
+		w.modTime = info.ModTime()
```

Comment:
AFAIK it’s not documented that `w.f.Close()` won’t update the timestamp (e.g. it could be buffering writes for some reason).

OTOH, sure, `Stat()` after closing the file may lose updates. But then all of this protects against ~rogue external writers who don’t lock correctly…

Reply:
I dunno how *documented* this is per se, but that is now how delayed writes work. The mtime information is part of the struct inode, and it will be modified when you do actual vfs operations, not when dirty buffers are flushed to disk.

Reply:
I was thinking of the Go `os.File` possibly buffering data, the way `<stdio.h>` traditionally does. I agree it would be surprising if it happened, but nothing explicitly promises that it won’t.

>  The mtime information is part of the struct inode, and it will be modified when you do actual vfs operations, not when dirty buffers are flushed to disk.

(BTW, https://pubs.opengroup.org/onlinepubs/9699919799/ “file times update” says that it might be delayed after the actual write even inside the kernel — in a way that wouldn’t matter here.)

Reply:
I mean, it better have pushed all os.File data to the kernel, or the fdatasync() call we just did was completely wasted, having persisted to disk a partial json file...

Anyway, I'm pretty sure that in practice (i.e. on linux) this will work. Even if Xenix 0.0.1 might do something weird and still be according to spec. And anwyay, as you say, if we get the wrong mtime all that happens is that we unnecessarily load the file again.

Reply:
> I mean, it better have pushed all os.File data to the kernel, or the fdatasync() call we just did was completely wasted,

Oh, that’s a strong argument.